runtime/lua/vim/lsp/rpc.lua

```diff
@@ -49,70 +49,102 @@ local header_start_pattern = ('content'):gsub('%w', function(c)
   return '[' .. c .. c:upper() .. ']'
 end)
 
+
+local has_strbuffer, strbuffer = pcall(require, "string.buffer")
```

Comment:
Could we shim `string.buffer` for PUC Lua? Even if it doesn't help performance there, it would simplify the code here and encourage more use of `string.buffer`.

Reply:
> if chatgpt is correct,

Not even close to correct, but can look into it.

Reply:
Looks like with puc lua 5.1 we can't support `#buf` because `__len` doesn't work on tables.
Might be possible to implement it in c and expose as userdata where `__len` afaik should work.

Reply:
Even if the shim just applies to this module it would mean we don't need 2 versions of the same function.

Reply:
Had a quick look at this but a lua shim would imho make it even more complicated. Emulating skip and the consume behavior of `:get`  would either be slower than before or more complex than the existing solution. 

I'd either keep the PR as is or just close this.